src/commands/Tools/create-emoji.ts

```diff
@@ -0,0 +1,43 @@
+import { LanguageKeys } from '#lib/i18n/languageKeys';
+import { SkyraCommand } from '#lib/structures';
+import { PermissionLevels } from '#lib/types/Enums';
+import { fetch, FetchResultTypes } from '#utils/util';
+import { ApplyOptions } from '@sapphire/decorators';
+import { EmojiRegex } from '@sapphire/discord.js-utilities';
+import { Args } from '@sapphire/framework';
+import { Message } from 'discord.js';
+
+@ApplyOptions<SkyraCommand.Options>({
+	aliases: ['add-emoji'],
+	generateDashLessAliases: true,
+	cooldown: 10,
+	permissionLevel: PermissionLevels.Moderator,
+	permissions: ['MANAGE_EMOJIS'],
+	description: LanguageKeys.Commands.Tools.CreateEmojiDescription,
+	extendedHelp: LanguageKeys.Commands.Tools.CreateEmojiExtended
+})
+export class UserCommand extends SkyraCommand {
+	public async run(message: Message, args: SkyraCommand.Args) {
+		const emojiData = await args.pick(UserCommand.emojiResolver);
+
+		try {
+			const emojiBuffer = await fetch(`https://cdn.discordapp.com/emojis/${emojiData.id}.png`, FetchResultTypes.Buffer);
+			const emoji = await message.guild!.emojis.create(emojiBuffer, emojiData.name);
+			return message.sendTranslated(LanguageKeys.Commands.Tools.CreateEmojiSuccess, [{ emoji: emoji.toString() }]);
+		} catch {
+			return this.error(LanguageKeys.Commands.Tools.CreateEmojiInvalidEmoji);
+		}
+	}
+
+	public static emojiResolver = Args.make<EmojiData>((parameter, { argument }) => {
+		const match = EmojiRegex.exec(parameter);
+		return match
+			? Args.ok({ id: match[3], name: match[2] })
```

Comment:
- Reference named capture groups instead of match[x]
- first check if groups are available before doing Args.ok

Reply:
`EmojiRegex` doesn't have a -WithGroups variant.

Reply:
That's not a variant. That's just the regex as is. It's even in the TSDoc of the regex:

```ts
/**
 * Regex that can capture the ID of any animated or non-animated custom Discord emoji
 * @raw `/^(?:<(?<animated>a)?:(?<name>\w{2,32}):)?(?<id>\d{17,21})>?$/`
 * @remark Capture group 1 can be used to determine whether the emoji is animated or not. It is named `animated`.
 * @remark Capture group 2 is the name of the emoji as it is typed in a message. It is named `name`.
 * @remark Capture group 3 is the ID of the emoji. It is named `id`.
 */
export const EmojiRegex = /^(?:<(?<animated>a)?:(?<name>\w{2,32}):)?(?<id>\d{17,21})>?$/;
```